lib/checks/mobile/css-orientation-lock.js

```diff
@@ -1,134 +1,251 @@
+					/orientation:\s*portrait/i.test(cssText)
+				);
+			})
+			/**
+			 * Filter:
+			 * Ensure RULELIST is not empty
+			 */
+			.filter(({ cssRules }) => cssRules.length > 0)
+	);
+}
 
-// extract styles for each orientation rule to verify transform is applied
-let isLocked = false;
-let relatedElements = [];
+/**
+ * Interpolate a given CSS Rule to ascertain if orientation is locked by use of any transformation functions that affect rotation along the Z Axis
+ * @param {Object} cssRule given CSS Rule
+ * @property {String} cssRule.selectorText selector text targetted by given cssRule
+ * @property {Object} cssRule.style style
+ * @return {Boolean}
+ */
+function getIsOrientationLocked({ selectorText, style }) {
+	if (!selectorText || style.length <= 0) {
+		return false;
+	}
 
-Object.keys(rulesGroupByDocumentFragment).forEach(key => {
-	const { root, rules } = rulesGroupByDocumentFragment[key];
+	const transformStyle =
+		style.transform || style.webkitTransform || style.msTransform || false;
+	if (!transformStyle) {
+		return false;
+	}
 
-	// filter media rules from all rules
-	const mediaRules = rules.filter(r => {
-		// doc: https://developer.mozilla.org/en-US/docs/Web/API/CSSMediaRule
-		// type value of 4 (CSSRule.MEDIA_RULE) pertains to media rules
-		return r.type === 4;
-	});
-	if (!mediaRules || !mediaRules.length) {
-		return;
+	/**
+	 * get last match/occurence of a transformation function that can affect rotation along Z axis
+	 */
+	const matches = transformStyle.match(
+		/(rotate|rotateZ|rotate3d|matrix|matrix3d)\(([^)]+)\)(?!.*(rotate|rotateZ|rotate3d|matrix|matrix3d))/
+	);
+	if (!matches) {
+		return false;
 	}
 
-	// narrow down to media rules with `orientation` as a keyword
-	const orientationRules = mediaRules.filter(r => {
-		// conditionText exists on media rules, which contains only the @media condition
-		// eg: screen and (max-width: 767px) and (min-width: 320px) and (orientation: landscape)
-		const cssText = r.cssText;
-		return (
-			/orientation:\s*landscape/i.test(cssText) ||
-			/orientation:\s*portrait/i.test(cssText)
-		);
-	});
-	if (!orientationRules || !orientationRules.length) {
-		return;
+	const [, transformFn, transformFnValue] = matches;
+	const degrees = getRotationInDegrees(transformFn, transformFnValue);
+	if (!degrees) {
+		return false;
 	}
 
-	orientationRules.forEach(r => {
-		// r.cssRules is a RULEList and not an array
-		if (!r.cssRules.length) {
-			return;
-		}
-		// cssRules ia a list of rules
-		// a media query has framents of css styles applied to various selectors
-		// iteration through cssRules and see if orientation lock has been applied
-		Array.from(r.cssRules).forEach(cssRule => {
-			// ensure selectorText exists
-			if (!cssRule.selectorText) {
-				return;
-			}
-			// ensure the given selector has styles declared (non empty selector)
-			if (cssRule.style.length <= 0) {
-				return;
-			}
+	const locked = degrees % 90 === 0 && degrees % 180 !== 0;
```

Comment:
We should allow for some margin of error. Especially when converting rad, we could be dealing with rounding errors. I think this needs to be an option in the check, how much of an error margin is on this rotation.